ros2topic/ros2topic/verb/pub.py

```diff
@@ -81,9 +99,18 @@ def publisher(
         node_name = NODE_NAME_PREFIX + '_publisher_%s' % (message_type.replace('/', '_'), )
     rclpy.init()
 
+    # Build a QoS profile based on user-supplied arguments
+    profile = rclpy.qos.qos_profile_system_default
+    if qos_profile:
+        profile = QOS_PROFILE_OPT[qos_profile]
```

Comment:
If you map `QOS_PROFILE_OPT[None]` to `rclpy.qos.qos_profile_system_default` then you could use just `profile = QOS_PROFILE_OPT` here without the conditional.